src/store/store.ts

```diff
@@ -50,8 +51,11 @@ export const useAppDispatch: () => AppDispatch = useDispatch;
 // This hook can be used in place of useDispatch to get correctly typed dispatches that return promises.
 export const usePromiseDispatch = () => {
   const dispatch = useAppDispatch();
-  return <Result>(action: AnyAction) =>
-    (dispatch as (action: AnyAction) => Promise<Result>)(action);
+  return useCallback(
+    <Result>(action: AnyAction) =>
+      (dispatch as (action: AnyAction) => Promise<Result>)(action),
+    [dispatch]
+  );
```

Comment:
This was causing the useEffect in the table component to rerun as it was recreating the function on each render.